generic3g/specs/FieldSpec.F90

```diff
@@ -1019,15 +1045,23 @@ recursive function make_adapters(this, goal_spec, rc) result(adapters)
       integer, optional, intent(out) :: rc
 
       integer :: status
+      type(VerticalGridAdapter) :: vertical_grid_adapter
 
       select type (goal_spec)
       type is (FieldSpec)
-         allocate(adapters(4))
+         allocate(adapters(5))
          allocate(adapters(1)%adapter, source=GeomAdapter(goal_spec%geom, goal_spec%regrid_param))
-         allocate(adapters(2)%adapter, &
-              source=VerticalGridAdapter(goal_spec%vertical_grid, goal_spec%geom, goal_spec%typekind, goal_spec%units, VERTICAL_REGRID_LINEAR))
-         allocate(adapters(3)%adapter, source=TypeKindAdapter(goal_spec%typekind))
-         allocate(adapters(4)%adapter, source=UnitsAdapter(goal_spec%units))
+         allocate(adapters(2)%adapter, source=VerticalDimSpecAdapter(goal_spec%vertical_dim_spec))
+         vertical_grid_adapter = VerticalGridAdapter( &
+              goal_spec%vertical_grid, &
+              goal_spec%vertical_dim_spec, &
+              goal_spec%geom, &
+              goal_spec%typekind, &
+              goal_spec%units, &
+              VERTICAL_REGRID_LINEAR)
```

Comment:
That's a lot of arguments.   I've not looked closely, but could we instead just pass `goal_spec` and "digest" it inside the constructor?